Scripts/Python/xHighLevelStarTrekDoor.py

```diff
@@ -144,219 +129,85 @@ def OnSDLNotify(self,VARname,SDLname,playerID,tag):
-            #PtDebugPrint("xHighLevelStarTrekDoor: Timer set to : %d" % self.respondertime)
-            #PtAtTimeCallback(self.key,self.respondertime,1)
-            self.ExecCode(code)
-            if self.DoorStack[0].find('fastforward=1') != -1:
-                self.DoorStack.pop(0)
+            elif self.DoorState == doorSDLstates['movingclosed'] or self.DoorState == doorSDLstates['closing']:
+                self.UpdateDoorState(doorSDLstates['closed'])
 
     ##########################################
-    def UpdateDoorState (self, StateNum):
+    def UpdateDoorState (self, StateNum, playerID = -1):
         if StateNum != self.DoorState:
             ageSDL = PtGetAgeSDL()
             self.DoorState = StateNum
             self.SDL['DoorState'] = (StateNum,)
-            self.SendNote('DoorState='+str(StateNum))
-            PtClearTimerCallbacks(self.key)
-            PtAtTimeCallback(self.key, 30, 1)
+            localPlayerID = PtGetLocalPlayer().getPlayerID()
             if self.DoorEnabled == 0:
-                return            
-
-            if self.DoorState == doorSDLstates['opening']:
-                self.SendNote("respOpenDoor;0")
-                PtDebugPrint("xHighLevelStarTrekDoor: Notifying Clients to play Open Door Responder")
-
-            elif self.DoorState == doorSDLstates['closing']:
-                self.SendNote("respCloseDoor;0")
-                PtDebugPrint("xHighLevelStarTrekDoor: Notifying Clients to play Close Door Responder")
-
-            elif self.DoorState == doorSDLstates['open']:
-                PtDebugPrint("xHighLevelStarTrekDoor: Updating Age SDL to Open")
-                ageSDL[strDoorClosedVar.value] = (0,)
-
-            elif self.DoorState == doorSDLstates['closed']:
-                PtDebugPrint("xHighLevelStarTrekDoor: Updating Age SDL to Closed")
-                ageSDL[strDoorClosedVar.value] = (1,)
-
-    ##########################################
-    # This is a cheap hack to attempt to get 
-    # the doors reset, should the age owner 
-    # screw up and not update the door states
-    ##########################################
-    def OnTimer(self,TimerID):
-        if self.sceneobject.isLocallyOwned():
-            PtDebugPrint("xHighLevelStarTrekDoor: Timer Came Back")
-            if self.DoorState == doorSDLstates['opentoclose']:
-                self.UpdateDoorState(doorSDLstates['closing'])
-
-            elif self.DoorState == doorSDLstates['movingopen'] or self.DoorState == doorSDLstates['opening']:
-                self.UpdateDoorState(doorSDLstates['open'])
-
-            elif self.DoorState == doorSDLstates['closetoopen']:
-                self.UpdateDoorState(doorSDLstates['opening'])
-
-            elif self.DoorState == doorSDLstates['movingclosed'] or self.DoorState == doorSDLstates['closing']:
-                self.UpdateDoorState(doorSDLstates['closed'])
+                return
+            if int(playerID) == int(localPlayerID):
```

Comment:
```suggestion
            if playerID == localPlayerID:
```

These should already be integers.